frontend/src/component/common/Sticky/StickyProvider.tsx

```diff
@@ -0,0 +1,133 @@
+                    return newItems.sort((a, b) => {
+                        const elementA = a.current;
+                        const elementB = b.current;
+                        if (elementA && elementB) {
+                            return (
+                                elementA.getBoundingClientRect().top -
+                                elementB.getBoundingClientRect().top
+                            );
+                        }
+                        return 0;
+                    });
+                }
+
+                return prevItems;
+            });
+        },
+        [],
+    );
+
+    const unregisterStickyItem = useCallback(
+        (ref: RefObject<HTMLDivElement>) => {
+            unregisterResizeListener(ref);
+            setStickyItems((prev) => prev.filter((item) => item !== ref));
+        },
+        [],
+    );
+
+    const registerResizeListener = useCallback(
+        (ref: RefObject<HTMLDivElement>) => {
+            setResizeListeners((prev) => new Set(prev).add(ref));
+        },
+        [],
+    );
+
+    const unregisterResizeListener = useCallback(
+        (ref: RefObject<HTMLDivElement>) => {
+            setResizeListeners((prev) => {
+                const newListeners = new Set(prev);
+                newListeners.delete(ref);
+                return newListeners;
+            });
+        },
+        [],
+    );
+
+    const getTopOffset = useCallback(
+        (ref: RefObject<HTMLDivElement>) => {
+            if (!stickyItems.some((item) => item === ref)) {
+                // Return 0 in case the item is not registered yet
+                return 0;
+            }
+            const stickyItemsUpToOurItem = stickyItems.slice(
+                0,
+                stickyItems.findIndex((item) => item === ref),
+            );
+            return stickyItemsUpToOurItem.reduce((acc, item) => {
+                if (item === ref) {
+                    // We should not include the current item in the calculation
+                    return acc;
+                }
+
+                // Accumulate the height of all sticky items above our item
+                const itemHeight =
+                    item.current?.getBoundingClientRect().height || 0;
+
+                return acc + itemHeight;
+            }, 0);
+        },
+        [stickyItems],
+    );
+
+    useEffect(() => {
+        const resizeObserver = new ResizeObserver(() => {
+            // We should recalculate top offsets whenever there's a resize
+            // This will trigger the dependency in `getTopOffset` and recalculate the top offsets in the Sticky components
+            setStickyItems((prev) => [...prev]);
+        });
+
+        resizeListeners.forEach((item) => {
+            if (item.current) {
+                resizeObserver.observe(item.current);
+            }
+        });
+
+        return () => {
+            if (resizeListeners.size > 0) {
+                resizeListeners.forEach((item) => {
```

Comment:
same here. Can we just iterate over the stickyItems?

Reply:
I don't think we should. It's important to keep a separate state for observables since our revalidation logic is based on re-setting `stickyItems`. If `stickyItems` is a dependency of this `useEffect` and we re-set it here, then we have infinite state updates.